python/cudf/cudf/api/types.py

```diff
@@ -230,13 +232,240 @@ def _union_categoricals(
     return cudf.Index(result_col)
 
 
+def is_bool_dtype(arr_or_dtype):
+    """
+    Check whether the provided array or dtype is of a boolean dtype.
+
+    Parameters
+    ----------
+    arr_or_dtype : array-like or dtype
+        The array or dtype to check.
+
+    Returns
+    -------
+    boolean
+        Whether or not the array or dtype is of a boolean dtype.
+
+    Examples
+    --------
+    >>> from cudf.api.types import is_bool_dtype
+    >>> import numpy as np
+    >>> import cudf
+    >>> is_bool_dtype(str)
+    False
+    >>> is_bool_dtype(int)
+    False
+    >>> is_bool_dtype(bool)
+    True
+    >>> is_bool_dtype(np.bool_)
+    True
+    >>> is_bool_dtype(np.array(['a', 'b']))
+    False
+    >>> is_bool_dtype(cudf.Series([1, 2]))
+    False
+    >>> is_bool_dtype(np.array([True, False]))
+    True
+    >>> is_bool_dtype(cudf.Series([True, False], dtype='category'))
+    True
+    """
+    if isinstance(arr_or_dtype, cudf.BaseIndex):
+        return arr_or_dtype._is_boolean()
+    elif isinstance(arr_or_dtype, cudf.Series):
+        if isinstance(arr_or_dtype.dtype, cudf.CategoricalDtype):
+            return is_bool_dtype(arr_or_dtype=arr_or_dtype.dtype)
```

Comment:
Should this be checking the boolean dtype of the categories?

```suggestion
            return is_bool_dtype(arr_or_dtype=arr_or_dtype.dtype.categories)
```

Reply:
I made a branch for `CategoricalDtype` at line `278` that will do this step while also generalizing for our `CategoricalDtype`.